extensions/dicom-seg/src/components/SegmentationPanel.js

```diff
@@ -0,0 +1,473 @@
+
+const { studyMetadataManager } = utils;
+
+/* TODO: wrap get/set interactions with the cornerstoneTools
+ * store with context to make these kind of things less blurry. */
+const segmentationModule = cornerstoneTools.getModule('segmentation');
+const { configuration } = segmentationModule;
+const DEFAULT_BRUSH_RADIUS = configuration.radius || 10;
+
+const refreshViewport = () => {
+  cornerstone.getEnabledElements().forEach(enabledElement => {
+    cornerstone.updateImage(enabledElement.element);
+  });
+};
+
+/**
+ * SegmentationPanel component
+ *
+ * @param {Object} props
+ * @param {Array} props.studies
+ * @param {Array} props.viewports - viewportSpecificData
+ * @param {number} props.activeIndex - activeViewportIndex
+ * @returns component
+ */
+const SegmentationPanel = ({ studies, viewports, activeIndex }) => {
+  const [brushRadius, setBrushRadius] = useState(DEFAULT_BRUSH_RADIUS);
+
+  /* TODO: We shouldn't hardcode this color, in the future the SEG may set the colorLUT to whatever it wants. */
+  const [brushColor, setBrushColor] = useState('rgba(221, 85, 85, 1)');
+  const [selectedSegment, setSelectedSegment] = useState();
+  const [showSegSettings, setShowSegSettings] = useState(false);
+  const [selectedSegmentation, setSelectedSegmentation] = useState();
+
+  const viewport = viewports[activeIndex];
+  const firstImageId = _getFirstImageId(viewport);
+  const { studyInstanceUid, seriesInstanceUid } = viewport;
+
+  /* CornerstoneTools */
+  const [brushStackState, setBrushStackState] = useState(
+    segmentationModule.state.series[firstImageId]
+  );
+
+  useEffect(() => {
+    setBrushStackState(segmentationModule.state.series[firstImageId]);
+  }, [studies, viewports, activeIndex, firstImageId]);
+
+  useEffect(() => {
+    if (brushStackState) {
+      setSelectedSegmentation(brushStackState.activeLabelmapIndex);
+    }
+
+    const labelmapModifiedHandler = event => {
+      console.warn('labelmap modified', event);
+      setBrushStackState(segmentationModule.state.series[firstImageId]);
+    };
+
+    /*
+     * These are specific to each element;
+     * Need to iterate cornerstone-tools tracked enabled elements?
+     * Then only care about the one tied to active viewport?
+     */
+    cornerstoneTools.store.state.enabledElements.forEach(enabledElement =>
+      enabledElement.addEventListener(
+        'cornersontetoolslabelmapmodified',
+        labelmapModifiedHandler
+      )
+    );
+
+    document.addEventListener('side-panel-change', handleSidePanelChange);
+
+    return () => {
+      cornerstoneTools.store.state.enabledElements.forEach(enabledElement =>
+        enabledElement.removeEventListener(
+          'cornersontetoolslabelmapmodified',
+          labelmapModifiedHandler
+        )
+      );
+
+      document.removeEventListener('side-panel-change', handleSidePanelChange);
+    };
+  });
+
+  const handleSidePanelChange = () => {
+    setShowSegSettings(false);
+  };
```

Comment:
Is this basically an "unmount" to reset the sidepanels state when we close/open? Basically, making sure when we open, segSettings is always hidden.

Reply:
Yes. Maybe we can find another way to do this?

Reply:
We could make sure panel component content receives access to the open/active props? Then it can react to those changes more directly?

Reply:
Event removed, now its directly using props from viewer.